packages/core/lib/commands/migrate/runMigrations.js

```diff
@@ -1,6 +1,110 @@
+const { createReadStream } = require("fs");
+const { readdir } = require("fs/promises");
+const path = require("path");
+const JSONStream = require("JSONStream");
 const Migrate = require("@truffle/migrate").default;
+const TruffleError = require("@truffle/error");
+const os = require("os");
+const debug = require("debug")("migrate:run");
+
+// Search for Push1 (60) to Push(32) 7F + console.log
+//                      60 ---- 7F  C O N S O L E . L O G
+const consoleLogRex = /[67][0-9a-f]636F6e736F6c652e6c6f67/i;
+
+async function usesConsoleLog(artifactJson) {
+  const debugLog = debug.extend("test");
+  debugLog("Artifact: %o", artifactJson);
+
+  //create a parser to get the value of jsonpath .deployedBytecode
+  const parser = JSONStream.parse(["deployedBytecode"]);
+  const stream = createReadStream(artifactJson).pipe(parser);
+
+  return new Promise((resolve, reject) => {
+    stream.on("data", data => {
+      //JSONParse will emit the entire string/value
+      //so initiate stream cleanup here
+      stream.destroy();
+      const usesConsoleLog = consoleLogRex.test(data);
+      debugLog("usesConsoleLog:", usesConsoleLog);
+      resolve(usesConsoleLog);
+    });
+
+    stream.on("error", err => {
```

Comment:
I think that would be a pretty non-standard implementation for this kind of a thing :/ I would vote we stick with the textbook usage described by Node in this case. Any other opinions?

Reply:
This is how this API works. The `error` event rejects the promise, and the `data` event resolves a bool.

Reply:
But it seems plausible that you could add a `try/catch` in the "data" callback and handle everything there. I don't really like it as it kind of feels like you are subverting the API. Does that miss any other errors that this might catch?

Reply:
An errors that occur in the data callback  function are not related to the stream and the stream won't know about those.

The pattern used here is correct.